hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconUtils.java

```diff
@@ -326,16 +353,19 @@ public static String constructFullPath(String keyName, long initialParentId, Str
       if (nsSummary == null) {
         log.warn("NSSummary tree is currently being rebuilt or the directory could be in the progress of " +
             "deletion, returning empty string for path construction.");
-        return "";
+        return fullPath;
```

Comment:
Looks like `TestNSSummaryEndpointWithFSO.testConstructFullPathWithNegativeParentIdTriggersRebuild` needs to be updated, too.